rubyforgood/human-essentials
# 3992 preserve storage location when editing purchase

Resolves #3992  

### Description

Introduced condition to determine whether a new purchase is being created or if an existing purchase is being edited. If the latter is true the current storage location is displayed rather than defaulting to the organizations storage location/first item on the drop-down.


### Type of change



* Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?


Visual Test:

1. Log in as an Organization Admin.
2. Go to the Purchases page (http://127.0.0.1:3000/diaper_bank/purchases).
3. View a purchase that has a Storage Location of 'Pawnee Main Bank (Office)'.
4. Click the 'Make a Correction' button.
5. Confirm that the Storage Location is correct and that it is not set to the first item in the drop-down list.

Rspec Test:
Added a test to spec/requests/purchases_requests_spec.rb to confirm that storage location is preserved editing a purchase and that the organization's storage location default is selected when creating a purchase.


### Screenshots

Before: 
![image](https://github.com/rubyforgood/human-essentials/assets/44501054/7fc8af9d-cf95-4260-bb48-0444703a23d4)

After:
![image](https://github.com/rubyforgood/human-essentials/assets/44501054/b9d48208-72e5-426a-b2e4-81c84c949b6e)
